docs/technical.rst

```diff
@@ -95,8 +95,26 @@ Features supported, encouraged and discouraged from the different ``JavaScript``
 TypeScript
 ^^^^^^^^^^
 
-``TODO``
+TypeScript is a type-safe version of JavaScript that compiles down to JavaScript. Type-safety
+offers a better development experience and improves the reliability and maintainability of
+large projects. TypeScript also produces type definition files to allow upstream projects to
+safely use EthereumJS libraries.
+
+TypeScript language support is a compile-time feature separate from the runtime JavaScript support.
+All EthereumJS projects that are written in TypeScript are compiled using TypeScript 3.2 or newer.
```

Comment:
Can we point this out more clearly, maybe as a separate isolated list item or something, what ``TypeScript`` version we currently use? This is one of the main information pieces people probably often want to get from this doc section and then people doesn't have to read this out from somewhere within the text.

Reply:
The point releases should technically be interchangeable\* so what's important would the what major version of TypeScript we're using: 3.

\* For the most part, they do include small breaking changes occasionally

Reply:
I'll update this to say TypeScript 3